### PR TITLE
Add tokens transfer ui tests

### PR DESCRIPTION
**Pull request by lyubenkov**
UI tests check tokens transfer between Solana and Neon EVMs.

Screenshot 2023-06-23 at 16 31 38
image

### CODE REVIEW

File: ui/conftest.py [hunk 1]
```diff
@@ -25,15 +25,20 @@
...
-    parser.addoption("--network", action="store", default="devnet", help="Which stand use")
-
-
 @pytest.fixture(scope="session")
 def network(pytestconfig: tp.Any) -> tp.Optional[str]:
     return EVM_NETWORKS.get(pytestconfig.getoption("--network"), EVM_NETWORKS["devnet"])
 
 
+@pytest.fixture(scope="session")
+def neonpass_url(pytestconfig: tp.Any) -> tp.Optional[str]:
```
Comment (gigimon) [inline ui/conftest.py:30]: I think we could move to top level conftest.py from integration directory and don't repeat common init code

File: ui/tests/test_neonpass.py [hunk 2]
```diff
@@ -72,15 +65,17 @@ def context(
 
 
 @pytest.fixture
-def request_sol() -> None:
+def request_sol(solana_url: str) -> None:
```
Comment (gigimon) [inline ui/tests/test_neonpass.py:64]: this already handled good in integration tests

**Review by gigimon — CHANGES_REQUESTED** [review]

File: ui/conftest.py (hunk 1 above)
Comment (lyubenkov) [reply ui/conftest.py:30]: Do you mean to move the `operator` fixture from integration directory to the root conftest file, and also include `neonpass_url` into it?

Comment (gigimon) [reply ui/conftest.py:30]: @lyubenkov in integration we already have code to initialize environment, read envs.json and more. In UI directory we do the same. Common parts of code move to root, specific left in integration and ui

Comment (lyubenkov) [reply ui/conftest.py:30]: Discussed and came to agreement, that we don't need to change anything here as it's already great!

File: ui/tests/test_neonpass.py (hunk 2 above)
Comment (lyubenkov) [reply ui/tests/test_neonpass.py:64]: The same as in another thread.

File: ui/conftest.py [hunk 3]
```diff
@@ -15,44 +16,46 @@
...
 
-CHROME_EXT_DIR = "extensions/chrome/plugins"
-"""Relative path to Chrome extension source
-"""
 
-CHROME_USER_DATA_DIR = "user_data"
-"""Relative path to Chrome extensions user data
-"""
+@pytest.fixture(scope="session")
+def network(pytestconfig: tp.Any) -> tp.Optional[str]:
```
Comment (gigimon) [inline ui/conftest.py:25]: we can't run this tests on night-stand

File: ui/conftest.py [hunk 4]
```diff
@@ -15,44 +16,46 @@
...
         else:
-            path = path / f",{source}"
-    return path
+            result_path = result_path / f",{source}"
+    yield result_path
+    libs.rm_tree(CHROME_DATA_PATH)
 
 
-@pytest.fixture(scope="session", autouse=True)
+@pytest.fixture(scope="function", autouse=True)
```
Comment (gigimon) [inline ui/conftest.py:55]: why you change this to function? It's slower and we don't need to unpack data for every test

File: ui/libs/__init__.py [hunk 5]
```diff
@@ -19,11 +21,24 @@ class Token:
     name: str
     address: str = None
 
+    def __str__(self):
+        return self.name
+
+
+@dataclass
+class EVM:
```
Comment (gigimon) [inline ui/libs/__init__.py]: bad name

File: ui/libs/__init__.py [hunk 6]
```diff
@@ -35,6 +50,19 @@ class Tokens:
 """
 
 
+@staticmethod
```
Comment (gigimon) [inline ui/libs/__init__.py]: @staticmethod applies only for methods in class

File: ui/libs/__init__.py [hunk 7]
```diff
@@ -69,6 +97,14 @@ def clone_user_data(extensions_dir: pathlib.Path) -> pathlib.Path:
     )
 
 
+def extract_tar_gz(source: pathlib.Path, dest: pathlib.Path) -> pathlib.Path:
+    """Extract source into destination"""
+    file = tarfile.open(source)
```
Comment (gigimon) [inline ui/libs/__init__.py]: tarfile support work use "with" operator, please use it

**Review by gigimon — APPROVED** [review]

File: ui/conftest.py (hunk 4 above)
Comment (lyubenkov) [reply ui/conftest.py:55]: This is made on purpose — both Phantom and MM plugins require wallet unlock step, but if we don't clean `chrome_data` directory, the next time wallets will be unlocked, which leads to inconsistent behaviour. To support the same initial conditions for all the tests I do `chrome_data` cleaning before each one. Please pay attention, that it doesn't apply to `plugins` directory, which is preserved between the tests — so basically I reset only the cache.

I suppose that in terms of the whole scope of UI tests this clean&untar procedure takes quite negligible time. Also if we parallel each test (parametrised) in future, then the wallet unlock step will be required before each one anyway.

File: ui/conftest.py (hunk 3 above)
Comment (lyubenkov) [reply ui/conftest.py:25]: If we set the `network` parameter in environment to the `night-stand` value, this method will return `NEON EVM night-stand` value (see the screenshot below). Currently I set  the neonpass_url parameter in the `envs.json` only for devnet environment, but it's also possible to set it for the `night-stand` as well.
Screenshot 2023-06-26 at 11 05 29
Screenshot 2023-06-26 at 11 28 28

File: ui/libs/__init__.py (hunk 6 above)
Comment (lyubenkov) [reply ui/libs/__init__.py]: Good caught, I removed that annotation!

File: ui/libs/__init__.py (hunk 7 above)
Comment (lyubenkov) [reply ui/libs/__init__.py]: Good idea! I added `with` statement!

File: ui/libs/__init__.py (hunk 5 above)
Comment (lyubenkov) [reply ui/libs/__init__.py]: Changed it to `Platform`, I hope it better reflects the meaning, but let me know if there's greater naming.

**Pull request merged**